pkg/metric/kubelet_stats_summary_collector.go

```diff
@@ -90,7 +90,8 @@ func (c *kubeletStatsSummaryCollector) Update(ch chan<- prometheus.Metric) error
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
 		body, _ := io.ReadAll(resp.Body)
-		return fmt.Errorf("failed to get kubelet stats summary: %d: %q", resp.StatusCode, string(body))
+		klog.V(4).InfoS("Failed to get kubelet stats summary", "status_code", resp.StatusCode, "body", string(body))
```

Comment:
Should the klog level be set to 2 to prevent excessive logging of infos?

Reply:
v=2 is the default level set in helm.

I think this is not the correct way to hide the error. Maybe we should add a flag to disable this collector completely?

Reply:
Added a featuregate for the collector